macrotranslator.py

```diff
@@ -0,0 +1,87 @@
+from macros import MacroMap, Macro, Invocation
+from translationconfig import TranslationConfig
+import logging
+import re
+
+logger = logging.getLogger(__name__)
+
+
+class MacroTranslator:
+    def __init__(self, translation_config: TranslationConfig) -> None:
+        self.translation_config = translation_config
+
+    def generate_macro_translations(self,
+                                    mm: MacroMap) -> dict[Macro, str | None]:
+        translationMap: dict[Macro, str | None] = {}
```

Comment:
```suggestion
        translationMap: dict[Macro, Optional[str]] = {}
```

You get the idea; not commenting on the later occurrences but you can change those as well.